Realm/RLMProperty.mm

```diff
@@ -343,6 +343,9 @@ - (instancetype)initSwiftPropertyWithName:(NSString *)name
     }
     else if ([_objcRawType isEqualToString:@"@\"NSNumber\""]) {
         const char *numberType = [propertyValue objCType];
+        if (!numberType) {
+            @throw RLMException(@"Can't persist NSNumber directly: use a Swift-native number type.");
```

Comment:
Oh, I see. What type do you suggest it default to? I'd presume a floating-point representation would be best so no unexpected truncations will occur.

Is there any way we can provide proper error messages to Realm Swift users while maintaining backwards compatibility for Realm Objective-C users with Swift models?